### PR TITLE
FUSETOOLS-2462 - integrating Context help on Global Config page

### PR DESCRIPTION
**Pull request by bfitzpat**
Signed-off-by: Brian Fitzpatrick <bfitzpat@redhat.com>

-- new context-ids will be enabled with content with upcoming help
update

# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [ ] Did you use the Jira Issue number in the commit comments?
- [ ] Did you set meaningful commit comments on each commit?
- [ ] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [ ] Did the CI job report a successful build?
- [ ] Is the non-happy path working, too?

## Maintainability

- [ ] Are all Sonar reported issues fixed or can they be ignored?
- [ ] Is there no duplicated code?
- [ ] Are method-/class-/variable-names meaningful?

## Tests

- [ ] Are there unit-tests?
- [ ] Are there integration tests (or at least a jira to tackle these)?
- [ ] Do we need a new UI test?

## Legal

- [ ] Have you used the correct file header copyright comment?
- [ ] Have you used the correct year in the headers of new files?

### CODE REVIEW

**Comment by bfitzpat** [comment]
May have muddled the desired behavior a bit here, so @MelissaFlinn will be updating FUSETOOLS-2462 with details and I'll take another stab at this on Monday.

**Comment by bfitzpat** [comment]
@lhein ready for review

**Comment by bfitzpat** [comment]
testPR

**Comment by bfitzpat** [comment]
Build seems to be failing due to a SpringBoot BOM issue - 
`Error Message

expected:<"2.2.170.redhat-0000[24]"> but was:<"2.2.170.redhat-0000[19]">

Stacktrace

org.junit.ComparisonFailure: expected:<"2.2.170.redhat-0000[24]"> but was:<"2.2.170.redhat-0000[19]">
	at org.fusesource.ide.camel.model.service.core.util.CamelCatalogUtilsTest.testBomVersionRetrieval(CamelCatalogUtilsTest.java:61)
`

**Comment by apupier** [comment]
> Build seems to be failing due to a SpringBoot BOM issue

I think the branch needs to be rebased on master

**Comment by bfitzpat** [comment]
Thanks for the clean-up @lhein

**Comment by lhein** [comment]
I would review it but it seems dirty as I was involved too much in the development of this PR :D @apupier

**Comment by apupier** [comment]
also noticed that when selecting a log component in design view, it is not the doc for doc that is displayed. Needs to check if it was already the case before. If yes, just open an issue, if no needs to be fixed in this PR.

**Comment by bfitzpat** [comment]
@apupier Fixed NPE you noted and provided a default context Id should we not be able to find one for the endpoint, bean, or data format.

**Comment by lhein** [comment]
I added the fix for the log component to this PR.

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/DocumentationSection.java [hunk 1]
```diff
@@ -117,11 +123,45 @@ protected void showDocumentationPage() {
...
+		// lets see if we can find the docs for an endpoints URI...
+		if (node.isEndpointElement()) {
+			contextId = getEndpointDocumentationPage();
+		} else if (node.getXmlNode() != null) {
+			Element element = (Element) node.getXmlNode();
+			String tagName = element.getTagName();
+			// if it's a bean, go with that context ID
+			if (BEAN_TAG.equals(tagName)) {
+				contextId = HELP_CONTEXT_ID_PREFIX + GLOBAL_BEAN_PAGE;
+			// if it's a data format, go with that.
```
[the remark this answers is not in the document]
Comment (lhein) [reply editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/DocumentationSection.java]: I can be also an endpoint or a sap configuration afaik.

[the remark this answers is not in the document]
Comment (bfitzpat) [reply editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/DocumentationSection.java]: Yes, you need https://github.com/jbosstools/jbosstools-fuse/pull/1477 with the additional data format help files in order to see help content. 

I don't think the log fix is going to hurt anything here.

And yes, in theory someone else could extend the global configuration framework we have in place. Unless it's a data format, the default GLOBAL_CONFIG_PAGE should appear (see https://github.com/jbosstools/jbosstools-fuse/pull/1476/files#diff-1c9486ce617b479184a99e98ea4d6ba4R136 ) which is why I included that as a default.

[the remark this answers is not in the document]
Comment (apupier) [reply editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/DocumentationSection.java]: my point was simply that the comment seems not accurate and wanted to ensure that it was just a comment issue and not an understanding of the overall Fuse Tooling code.

[the remark this answers is not in the document]
Comment (apupier) [reply editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/DocumentationSection.java]: > can be also a sap configuration

sap configuration are just specific beans, so it falls in previous if

**Pull request merged**